core/src/main/java/org/jdbi/v3/core/argument/EnumArgumentFactory.java

```diff
@@ -14,21 +14,57 @@
 package org.jdbi.v3.core.argument;
 
 import java.lang.reflect.Type;
+import java.sql.Types;
 import java.util.Optional;
+import java.util.function.Function;
 
-import org.jdbi.v3.core.argument.internal.strategies.LoggableBinderArgument;
 import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.enums.EnumStrategy;
+import org.jdbi.v3.core.generic.GenericTypes;
+import org.jdbi.v3.core.internal.EnumStrategies;
+import org.jdbi.v3.core.qualifier.QualifiedType;
 
-class EnumArgumentFactory implements ArgumentFactory {
+class EnumArgumentFactory implements QualifiedArgumentFactory {
     @Override
-    public Optional<Argument> build(Type expectedType, Object rawValue, ConfigRegistry config) {
-        // Enums must be bound as VARCHAR
-        // TODO use the same configuration as EnumMapperFactory for consistency
-        if (rawValue instanceof Enum) {
-            Enum<?> enumValue = (Enum<?>) rawValue;
-            return Optional.of(LoggableBinderArgument.bindAsString(enumValue.name()));
-        } else {
-            return Optional.empty();
-        }
+    @SuppressWarnings("unchecked")
+    public Optional<Argument> build(QualifiedType<?> givenType, Object value, ConfigRegistry config) {
+        return ifEnum(givenType.getType())
+            .flatMap(clazz -> makeEnumArgument((QualifiedType<Enum>) givenType, (Enum) value, config));
+    }
+
+    @SuppressWarnings("unchecked")
+    static <E extends Enum<E>> Optional<Class<?>> ifEnum(Type type) {
```

Comment:
can / should this be `Optional<Class<E>>`?

Reply:
It can. I'll change.